src/v/kafka/server/group_manager.h

```diff
@@ -179,6 +179,48 @@ class group_manager {
           , partition(std::move(p)) {}
     };
 
+    struct read_write_lw_lock {
```

Comment:
this is really a biased writer lock. i.e.: any reader will get preempted (non-fifo) by any active writer. 

so this has starvation properties, which I assumed are explicit by design. 

Can you make a comment on the bias, and why that's useful. Also, would be interesting in writing a comment as to why `seastar::rw_lock` wouldn't work here (relatively similar impl - need to double check)

Reply:
I'll rewrite to seastar::rw_lock wasn't aware about it, the contention is very low we take the writer lock only on re-election which doesn't happen often in a stable cluster